src/dirs.proj

```diff
@@ -19,11 +19,44 @@
     <Project Include="XMakeBuildEngine\UnitTestsPublicOM\Microsoft.Build.Engine.OM.UnitTests.csproj" />
   </ItemGroup>
 
+  <Import Project="dir.targets" />
   <Import Project="..\dir.traversal.targets" />
 
   <PropertyGroup>
+    <SkipVersionGeneration>true</SkipVersionGeneration>
+    <!-- Turn off the automated package generation in packages.targets:AddNuGetPackageVersionMetadataToNuspecs;
+         we'll use the one produced by NerdBank.GitVersioning instead. -->
+    <DoNotGeneratePackageVersion>true</DoNotGeneratePackageVersion>
+  </PropertyGroup>
+
+  <Import Project="$(ToolsDir)packages.targets" Condition="Exists('$(ToolsDir)packages.targets') and '$(ImportGetNuGetPackageVersions)' != 'false'" />
+
+  <PropertyGroup Condition="Exists('$(ToolsDir)packages.targets') and '$(ImportGetNuGetPackageVersions)' != 'false'">
+    <TraversalBuildDependsOn>
+      $(TraversalBuildDependsOn);
+      GetNuspecProperties;
+      BuildPackages;
+    </TraversalBuildDependsOn>
+    <!-- We use Microsoft.DotNet.Build tools to create our packages, but we want the version comes from 
+         NerdBack.GitVersioning. -->
+    <GetNuGetPackageVersionsDependsOn>SetPackageVersion</GetNuGetPackageVersionsDependsOn>
```

Comment:
`BeforeTargets="BuildPackages"` means "if you're going to run `BuildPackages`, ensure that you've already run this target first.

Both of the things you're trying to do are affecting only in-memory state of the current project, so I think my way will do the trick. Let me know if not :)